modules/nf-core/variancepartition/dream/templates/dream.R

```diff
@@ -172,28 +244,41 @@ fitmm <- eBayes(fitmm, proportion = opt\$proportion,
 head(fitmm\$design, 3)
 print(colnames(fitmm\$design))
 
+has_intercept <- "(Intercept)" %in% colnames(fitmm\$design)
+
 # If contrast_string is provided, use that for makeContrast
 if (!is.null(opt\$contrast_string)) {
-    cat("Using contrast string:", opt\$contrast_string, "\n")
+    contrast_string <- opt\$contrast_string
+} else if (has_intercept) {
+  contrast_string <- ""
+    # Intercept model → use coef (reference level is implicit)
+    coef_name <- paste0(opt\$contrast_variable, opt\$contrast_target)
+    cat("Using default contrast matrix:", coef_name, "\n")
+
+    results <- topTable(fitmm, coef = coef_name, number = Inf,
+                        adjust.method = opt\$adjust.method, p.value = opt\$p.value,
+                        lfc = opt\$lfc, confint = opt\$confint)
```

Comment:
Do we need this case? 
I thought we only accept a formula in combination with a contrast_string. 

If we built the formula within this script, we already know that it is a zero-intercept model. 

I think combining custom formula with (baseline, variable, target) tuples will open up too many edge cases. So rather simplify here and add a check for that.